pkg/healthv2/provider.go

```diff
@@ -0,0 +1,255 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package healthv2
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/cilium/cilium/pkg/healthv2/types"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/statedb"
+	"github.com/cilium/cilium/pkg/statedb/index"
+	"github.com/cilium/cilium/pkg/time"
+)
+
+var logger = logging.DefaultLogger.WithField(logfields.LogSubsys, "healthv2")
+
+type HealthProviderParams struct {
+	cell.In
+	Lifecycle cell.Lifecycle
+	DB        *statedb.DB
+}
+
+type HealthProvider struct {
+	db           *statedb.DB
+	stopped      atomic.Bool
+	StatusTable  statedb.RWTable[types.Status]
```

Comment:
I would not export `RWTable` but keep it private to the provider. 

I would also make the `Table[types.Status]` available separately from `*HealthProvider`, and also have the indexes as top-level variables instead embedded into the health provider. The idea being that we would try to not "hide" the read-only access to statedb tables as they're central to how components using them work. E.g. it's easier to understand the health stuff when you see the table and `types.Status`.